test/event_sampler.spec.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+
+const EVENT_SAMPLE_RATE = require('../ext/tags').EVENT_SAMPLE_RATE
+
+describe('eventSampler', () => {
+  let eventSampler
+  let span
+
+  beforeEach(() => {
+    eventSampler = require('../src/event_sampler')
+    span = {
+      context: sinon.stub().returns({
+        _name: 'web.request'
+      }),
+      setTag: sinon.spy()
+    }
+  })
+
+  describe('sample', () => {
+    it('should sample a span with the provided rate', () => {
+      eventSampler.sample(span, 0.5)
+
+      expect(span.setTag).to.have.been.calledWith(EVENT_SAMPLE_RATE, 0.5)
+    })
+
+    it('should sample a span with the operation specific rate', () => {
+      eventSampler.sample(span, {
+        'web.request': 0.5
```

Comment:
:+1: I like this object option for configuration

Reply:
The only thing I'm not sure is how this will work for GraphQL resolvers since there can be hundreds of `graphql.resolve` events for a single request.

Reply:
Let that be future Roch's problem.

The issue here is because the tag might not be only on the root span right?

Reply:
It's the operation name, and will apply at any resolver level. But then, if you want the resolver events, I would also expect that you want it for every resolver, so it's probably expected. The only problem would be high cost at that point given the number of spans.